waku/waku_rln_relay/protocol_types.nim

```diff
@@ -49,8 +53,21 @@ type RateLimitProof* = object
   ## nullifier enables linking two messages published during the same epoch
   ## see details in https://hackmd.io/tMTLMYmTR5eynw2lwK9n1w?view#Nullifiers
   nullifier*: Nullifier
-  ## Application specific RLN Identifier
-  rlnIdentifier*: RlnIdentifier
+  when defined(rln_v2):
+    ## the external nullifier used for the generation of the `proof` (derived from poseidon([epoch, rln_identifier]))
+    externalNullifier*: ExternalNullifier
+  else:
+    ## the epoch used for the generation of the `proof`
+    epoch*: Epoch
+    ## Application specific RLN Identifier
+    rlnIdentifier*: RlnIdentifier
+
+when defined(rln_v2):
+  type ExtendedRateLimitProof* = ref object of RateLimitProof
+    ## epoch is the epoch for which the proof is generated
+    epoch*: Epoch
+    ## rlnIdentifier is the identifier of the RLN application
+    rlnIdentifier*: RlnIdentifier
```

Comment:
The distinction between the two is made because now, `RateLimitProof` contains the proof with the public inputs/outputs. epoch and rlnIdentifier are no longer public inputs, and must not be confused this way. `ExtendedRateLimitProof` is what should be sent on the wire.